[CI] SLMSnapshotBlockingIntegTests.testSLMRetentionAfterRestore failure

**Build scan**:

https://gradle-enterprise.elastic.co/s/7qn5kbuvv52le

**Repro line**:

./gradlew ':x-pack:plugin:ilm:test' --tests "org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.testSLMRetentionAfterRestore" \
  -Dtests.seed=FC982EB81E7E317B \
  -Dtests.security.manager=true \
  -Dtests.locale=es-AR \
  -Dtests.timezone=Asia/Taipei \
  -Druntime.java=11

**Reproduces locally?**:

No

**Applicable branches**:

master

**Failure excerpt**:

```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[SnapshotInfo{snapshotId=snap-nrdpzrxorlsfo3lgnjowfg/lYLY8csgRbWBJ3P0BsShxQ, state=SUCCESS, reason='null', indices=[test], startTime=1595305966745, endTime=1595305967572, totalShards=1, successfulShards=1, includeGlobalState=true, version=8.0.0, shardFailures=[]}]>
	at __randomizedtesting.SeedInfo.seed([FC982EB81E7E317B:EDAE146EB0D0B1AB]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.lambda$testSLMRetentionAfterRestore$11(SLMSnapshotBlockingIntegTests.java:427)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:897)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:870)
	at org.elasticsearch.xpack.slm.SLMSnapshotBlockingIntegTests.testSLMRetentionAfterRestore(SLMSnapshotBlockingIntegTests.java:423)
```

This is possible related to https://github.com/elastic/elasticsearch/issues/59140, but the method failing here is `testSLMRetentionAfterRestore` and the error looks a bit different. Opening as a separate issue, but feel free to close if the two are connected.

--- a/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java
+++ b/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java
@@ -429,14 +429,7 @@ public class SLMSnapshotBlockingIntegTests extends AbstractSnapshotIntegTestCase
                 // This is what we want to happen
             }
             logger.info("--> snapshot [{}] has been deleted", snapshotName);
-        });
-
-        // Cancel/delete the snapshot
-        try {
-            client().admin().cluster().prepareDeleteSnapshot(REPO, snapshotName).get();
-        } catch (SnapshotMissingException e) {
-            // ignore
-        }
+        }, 30L, TimeUnit.SECONDS);
     }
 
     private SnapshotsStatusResponse getSnapshotStatus(String snapshotName) {

